setup: replace software selection widget with input dialog

incognitojam opened the pull request:
remove the "Choose Software to Install" screen and jump straight to inputting an installer URL after confirming wi-fi network.

|New|Old|
|-|-|
|Removed|![Choose Software to Install (custom)](https://user-images.githubusercontent.com/4038174/217147756-f0ef1714-fc98-42df-8ec7-dc0539a315d0.png)|



closes https://github.com/commaai/agnos-builder/issues/105

incognitojam committed:
setup: replace software selection widget with input dialog

adeebshihadeh commented on selfdrive/ui/qt/setup/setup.cc:
```diff
@@ -170,7 +170,16 @@ QWidget * Setup::network_setup() {
   cont->setProperty("primary", true);
-  QObject::connect(cont, &QPushButton::clicked, this, &Setup::nextPage);
+  QObject::connect(cont, &QPushButton::clicked, [=]() {
+    QString url = InputDialog::getText(tr("Enter installer URL"), this, "");
```
let's put back the old prompt here
```
Enter URL
for Custom Software
```

adeebshihadeh commented on selfdrive/ui/qt/setup/setup.cc:
```diff
@@ -170,7 +170,16 @@ QWidget * Setup::network_setup() {
+    if (url.isEmpty()) {
+      return;
+    }
+    setCurrentWidget(downloading_widget);
```
looks like this is slightly different. the logic here was carefully done to avoid glitches when opening/closing the input popup. did you test that?

incognitojam replied:
my bad, I see the problem

adeebshihadeh replied:
cool, setup specifically has to be super well tested since we don't use it everyday

incognitojam committed:
revert

https://github.com/commaai/openpilot/pull/21617/commits/3f7047651c31b209c9c4a0b8ff7bd408aca3f9e5

adeebshihadeh commented:
Can you open an agnos-builder PR to update the setup? You can just copy the one from your device in.
